app/controllers/hq/users_controller.rb

```diff
@@ -6,4 +6,24 @@ def index
     @users = User.all
   end
 
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      flash[:success] = 'User successfully saved'
+      redirect_to hq_user_url(@user)
+    else
+      render 'edit'
+    end
+  end
+
+private
+
+  def user_params
+    params[:user].permit(:user_name, :email)
```

Comment:
The only change I'm suggesting is
```ruby
def user_params
  params.require(:user).permit(:user_name, :email)
end
```
If the params hash is missing the `user` key, the controller will automatically raise the `ParameterMissing` error without even attempting to create or update a user record.